### PR TITLE
fix: limit the gas price of the mev bid

### PR DESCRIPTION
**Pull request by irrun**
### Description

fix: imit the gas price of the mev bid

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,8 +17,8 @@ require (
 	github.com/btcsuite/btcd/btcec/v2 v2.3.2
 	github.com/cespare/cp v1.1.1
 	github.com/cloudflare/cloudflare-go v0.79.0
-	github.com/cometbft/cometbft v0.37.0
 	github.com/cockroachdb/pebble v1.1.0
+	github.com/cometbft/cometbft v0.37.0
```
Comment (unclezoro) [inline go.mod:21]: reduce unnecessary changes

File: miner/bid_simulator.go [hunk 2]
```diff
@@ -78,6 +78,7 @@ type simBidReq struct {
 type bidSimulator struct {
 	config        *MevConfig
 	delayLeftOver time.Duration
+	minGasPrice   *big.Int
```
Comment (unclezoro) [inline miner/bid_simulator.go:81]: no need for this config. we reuse the config in the miner module.

File: miner/bid_simulator.go [hunk 3]
```diff
@@ -617,15 +621,35 @@ func (b *bidSimulator) simBid(interruptCh chan int32, bidRuntime *BidRuntime) {
...
-		return
+	// check if bid gas price is lower than min gas price
+	{
+		baseFee := bidRuntime.env.header.BaseFee
+		bidGasUsed := uint64(0)
+		bidGasFee := new(big.Int)
+
+		for i, receipt := range bidRuntime.env.receipts {
+			bidGasUsed += receipt.GasUsed
+			effectiveGasPrice := new(big.Int).Add(baseFee, bidRuntime.env.txs[i].EffectiveGasTipValue(baseFee))
```
Comment (unclezoro) [inline miner/bid_simulator.go:641]: what is the usage of `effectiveGasPrice`?

File: miner/bid_simulator.go (hunk 2 above)
Comment (irrun) [reply miner/bid_simulator.go:81]: it is

File: miner/bid_simulator.go (hunk 3 above)
Comment (irrun) [reply miner/bid_simulator.go:641]: for calculating txFee. If using the balance of system address, there maybe some transfer tx as tips to system.

Comment (unclezoro) [reply miner/bid_simulator.go:641]: yeah, that is what we want I think, sending BNB to system address is also valid.

Comment (irrun) [reply miner/bid_simulator.go]: fixed

**Review by unclezoro — APPROVED** [review]

**Review by zzzckck — APPROVED** [review]
